src/Jab/ContainerGenerator.cs

```diff
@@ -38,7 +38,14 @@ private void GenerateCallSiteWithCache(CodeWriter codeWriter, string rootReferen
                     });
             }
 
-            valueCallback(codeWriter, w => w.Append($"{cacheLocation}"));
+            if (serviceCallSite.ImplementationType.IsValueType)
+            {
+                valueCallback(codeWriter, w => w.Append($"{cacheLocation}!.Value"));
```

Comment:
Is `!` required? There is an `if ... == null` check right above.

Reply:
Technically, but not really, it's a == default check which still causes the compiler to throw an error, I can change it to == null and remove the !

Reply:
`== null` sounds good.